### ISSUE TITLE
Pervasive build failures on geminga (Trilinos' Version of Kokkos-Kernels)

### ISSUE THREAD

**Issue by csiefer2** [kokkos/kokkos-kernels#347]
A bunch of the print_specialization functions are only defined if KOKKOSKERNELS_ENABLE_TPL_BLAS is enabled, yet are used if
KOKKOSKERNELS_ENABLE_TPL_CUBLAS is enabled.

This leads to pervasive build failures.  If both of these really need to be enabled to make the code work, this mismatch should be detected at configure time.

See:
https://testing.sandia.gov/cdash/viewBuildError.php?buildid=4197910

**Comment by csiefer2** [comment]
As per @ndellingwood Kokkos-Kernels will not build if TPL_ENABLE_BLAS is disabled when CUBLAS is enabled.

**Comment by ndellingwood** [comment]
See these lines of code:

[CMakeLists.txt](https://github.com/kokkos/kokkos-kernels/blob/94456cf9099e3dcdd9bfb91a86a09a4f77a246d7/CMakeLists.txt#L293-L307)

**Comment by ndellingwood** [comment]
Suggested fix:

After this [line](https://github.com/kokkos/kokkos-kernels/blob/94456cf9099e3dcdd9bfb91a86a09a4f77a246d7/CMakeLists.txt#L300), enable BLAS if not enabled. That is 

```
IF(${Kokkos_ENABLE_Cuda})
IF (NOT KOKKOSKERNELS_ENABLE_TPL_BLAS)
  SET(KOKKOSKERNELS_ENABLE_TPL_BLAS ON)
  LIST(APPEND TPL_LIST "BLAS")
ENDIF()
...
```

**Comment by crtrott** [comment]
Solved in #348.
